### PR TITLE
Add intel-bootc multistage Containerfile

### PR DESCRIPTION
**Pull request by enriquebelarte**
Adds different steps for building required libraries, packages and dependencies for Intel Habanalabs on bootc image.

### CODE REVIEW

File: training/intel-bootc/Containerfile [hunk 1]
```diff
@@ -1,95 +1,163 @@
...
+    make \
+    gcc-c++ \
+    unzip \
+    && dnf clean all && rm -rf /var/cache/yum 
 
-COPY duplicated/common/usr /usr
+#Build ninja-build
+RUN git clone https://github.com/ninja-build/ninja.git \
+    && cd ninja \
+    && ./configure.py --bootstrap
```
Comment (rhatdan) [inline training/intel-bootc/Containerfile]: Does this get installed somewhere?  If yes can you remove ninja directory when done?

File: training/intel-bootc/Containerfile [hunk 2]
```diff
@@ -1,95 +1,163 @@
...
+    habanalabs-firmware-odm-${DRIVER_VERSION}.el9 \
+    && rm -f /etc/yum.repos.d/habanalabs.repo && rm -f /etc/yum.repos.d/habana.repo && \
+    dnf clean all && rm -rf /var/cache/yum \
+    && mv /tmp/extra /usr/lib/modules/${KERNEL_VERSION}.${TARGET_ARCH} \
+    && echo "softdep habanalabs post: habanalabs_ib" > /etc/modprobe.d/habanalabs_ib_dep.conf \
+    && depmod -a ${KERNEL_VERSION}.${TARGET_ARCH} \
+    && rm -rf tmp/*
+
+RUN python3.11 -m pip install pip==23.3.1 setuptools==67.3.3 wheel==0.38.4
+RUN python3.11 -m pip install habana_media_loader=="1.17.1.40"
```
Comment (rhatdan) [inline training/intel-bootc/Containerfile]: Can this line be merged with the line above?

File: training/intel-bootc/Containerfile [hunk 3]
```diff
@@ -1,95 +1,163 @@
...
     && export TARGET_ARCH=$(rpm -q --qf '%{ARCH}' kernel-core) \
-    && mv /tmp/extra /lib/modules/${KERNEL_VERSION}.${TARGET_ARCH} \
-    && mv /tmp/firmware/habanalabs /lib/firmware \
-    && depmod -a ${KERNEL_VERSION}.${TARGET_ARCH}
-
-RUN mv /etc/selinux /etc/selinux.tmp \
-    dnf install -y ${EXTRA_RPM_PACKAGES} \
+    && dnf -y update --exclude=kernel* --exclude=microcode_ctl \
+    && dnf install -y --nodocs --allowerasing --best \
+    libogg-devel \
```
Comment (rhatdan) [inline training/intel-bootc/Containerfile:111]: Are these devel rpms required in production or only during container build?

File: training/intel-bootc/Containerfile (hunk 1 above)
Comment (enriquebelarte) [reply training/intel-bootc/Containerfile]: Absolutely

File: training/intel-bootc/Containerfile [hunk 4]
```diff
@@ -1,95 +1,163 @@
...
+COPY --from=libbuilder /usr/lib/habanalabs/libhccl_ofi_wrapper.so /usr/lib/habanalabs/libhccl_ofi_wrapper.so
+COPY --from=libbuilder /opt/habanalabs/libfabric-1.20.0 /opt/habanalabs/libfabric-1.20.0
+COPY scripts/os_dependencies.sh /tmp/os_dependencies.sh
+
+#Install python3.11 and other build stuff
+RUN /tmp/os_dependencies.sh \
+    && dnf install -y \
+    python3.11 \
+    python3.11-pip \
+    python3.11-devel \
```
Comment (rhatdan) [inline training/intel-bootc/Containerfile:92]: Can we remove the devel package after the `make`?

File: training/intel-bootc/Containerfile (hunk 3 above)
Comment (enriquebelarte) [reply training/intel-bootc/Containerfile:111]: Just for building. Will remove them.

**Comment by rhatdan** [comment]
I will merge, but we still need to either move some of this devel work out of the finaly image and not committed in layers.

LGTM

**Pull request merged**